api/README.md

```diff
@@ -7,8 +7,84 @@ source .venv/bin/activate
+```
+
+2. Please also consider following these [instructions](https://sage-schematic.readthedocs.io/en/develop/index.html) to fill in configuration file and run the following command to obtain credential.json and token.pickle:
+```bash 
+schematic init --config ~/path/to/config.yml
+```
 
+Note: credentials.json and token.pickle have to be in the same directory as run_api.py. If you have obtained credentials.json before (but credentials.json is a different directory), please put config.yml and .synapseConfig in the same directory as run_api.py and run the above command again. 
+
+After running the command above, please follow instructions in your console and visit the URL to authorize schematic. After putting in the authorization code, token.pickle and credentials.json should be automatically downloaded in your local directory. 
+
+## Access Results
+Access the Swagger UI docs at this location:
 ```bash
 http://localhost:3001/v1/ui/
-```
\ No newline at end of file
+```
+
+## Notes for trouble shooting
+1. ImportError: cannot import name 'soft_unicode' from 'markupsafe' 
+2. ImportError: cannot import name 'json' from 'itsdangerous' 
+
+To resolve: 
+
+Make sure that the following libraries have the correct version: 
+* itsdangerous version: 1.1.0
+* jinja2 version: 2.11.3
+* markupsafe version: 2.0.1
+
+## Notes for using API endpoints: 
+Before trying out the API endpoints, please make sure that you have obtained `credentials.json`, `schematic_service_account_creds.json`, and `token.pickle`. 
+
+
+###  GET /manifest/generate
+
+This endpoint functions similarly like the following: 
+```bash
+schematic manifest -c ~/path/to/config.yml get -d <synapse id> -s -oauth
+```
+
+Examples: 
+1) Generate a patient manifest by using the sample data model.
+
+    Simply click on "Try it out" on swagger UI and then input "None" for dataset_id parameter
+
+2) Get an existing biospecimen manifest (synapse ID: syn26543611) from HTAN TNP-TMA by using HTAN data model: 
+
+* Step 1:  Make sure you have credentials to download the manifest from Synapse 
+
+    Try to visit [HTAN TNP-TMA](https://www.synapse.org/#!Synapse:syn26543611) and see if you could download the manifest from the UI. The "download" button should be disabled if you don't have credentials. 
+
+* Step 2: Update master_fileview in config.yml
+
+    Make sure that your config.yml is pointing to the right master_fileview. For HTAN, the master fileview should be: [syn20446927](https://www.synapse.org/#!Synapse:syn20446927/tables/)
+
+* Step 3: Use parent id of the manifest for "dataset_id" parameter
+
+    Make sure that you use the parent id instead of the id of the manifest. For example, the parent Id of the biospecimen manifest that we want to obtain is: [syn24995166](https://www.synapse.org/#!Synapse:syn24995166)
+
+To sum up, use the following values to fill out the parameters in swagger UI: 
+* schema_url: https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld
```

Comment:
Let's keep generic and not have references to HTAN.

Reply:
@milen-sage thank you! The updated version is more general now. I have removed the references to HTAN.